crates/node/p2p/src/rpc/request.rs

```diff
@@ -47,6 +66,130 @@ impl P2pRpcRequest {
         });
     }
 
+    fn handle_peers(
+        sender: Sender<PeerDump>,
+        connected: bool,
+        gossip: &GossipDriver,
+        disc: &Discv5Handler,
+    ) {
+        let Ok(total_connected) = gossip.swarm.network_info().num_peers().try_into() else {
+            error!(target: "p2p::rpc", "Failed to get total connected peers. The number of connected peers is too large and overflows u32.");
+            return;
+        };
+
+        let peer_ids: Box<dyn Iterator<Item = PeerId>> = if connected {
+            Box::new(gossip.swarm.connected_peers().cloned())
+        } else {
+            Box::new(gossip.peerstore.keys().cloned())
+        };
+        let peer_ids: Vec<PeerId> = peer_ids.collect();
```

Comment:
nit - can remove the need for the intermediate boxed trait object.
```suggestion
        let peer_ids: Vec<PeerId> = if connected {
            gossip.swarm.connected_peers().cloned().collect()
        } else {
            gossip.peerstore.keys().cloned().collect()
        };
```

Reply:
Very good suggestion!